app/models/warehouse/case_report.rb

```diff
@@ -94,18 +94,26 @@ def generate_all
         self.process_cases(Case::Base.all)
       end
 
-      def reconcile
+      def reconcile(job_size=nil)
         [
-          self.reconcile_missing_cases,
+          self.reconcile_missing_cases(job_size),
           self.reconcile_deleted_cases
         ]
       end
 
-      def reconcile_missing_cases
-        query =
-          Case::Base
-            .joins('LEFT OUTER JOIN warehouse_case_reports ON warehouse_case_reports.case_id = cases.id')
-            .where("cases.deleted = 'false' AND warehouse_case_reports.case_id IS NULL")
+      def reconcile_missing_cases(job_size=nil)
+        if job_size.nil?
+          query =
+            Case::Base
+              .joins('LEFT OUTER JOIN warehouse_case_reports ON warehouse_case_reports.case_id = cases.id')
+              .where("cases.deleted = 'false' AND warehouse_case_reports.case_id IS NULL")
+        else
+          query =
+            Case::Base
+              .joins('LEFT OUTER JOIN warehouse_case_reports ON warehouse_case_reports.case_id = cases.id')
+              .where("cases.deleted = 'false' AND warehouse_case_reports.case_id IS NULL")
+              .limit(job_size)
```

Comment:
I think having the main body of the query duplicated might be hard to maintain - it's the same but people might think it's meant to be different. Could we collapse the conditional and just call limit optionally if job_size is not nil?

Something like 

```suggestion
          query = Case::Base
              .joins('LEFT OUTER JOIN warehouse_case_reports ON warehouse_case_reports.case_id = cases.id')
              .where("cases.deleted = 'false' AND warehouse_case_reports.case_id IS NULL")
           query = query.limit(job_size) if job_size
```

Reply:
@Polsonby , I did this but somehow I got impression it will return the whole set first (which is the issue I am trying to avoid) then will apply limit afterwards,  let me try it again

Reply:
Yeah, it's hard to prove - the idea is that these _should_ create ActiveRecord Relation objects that don't hit the DB until the last possible moment, precisely so you can modify them iteratively. But it seems that in the console when you try it you always see it hit the DB because of the way the console wraps up the method calls in order to display the output according to "some guy on StackOverflow". Did you manage to verify the behaviour one way or the other?

Reply:
@Polsonby , tried both,  doesn't make any much different based on the figure from the measurement -_-! , this is why I made those changes based on your feedback,  I am only trusting the Ruby on rails should have considered it already,